Remove Rule board from docs

s-santillan opened the pull request:
# Thanks for improving Semgrep Docs 😀

### Please ensure

(N/A) A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs
- [x] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

----

 - Update all Rule board links because the endpoint is about to change (`.../board/` to `.../policies/` - see PR below)
- Update associated text (procedures, description of policies) because the flow is slightly different for various tasks.
- Set up redirect for docs-related links (`semgrep-code/rule-board` to `semgrep-code/policies/`) in docusaurus.config.js
- Retained mentions of rule board in Release notes.

For reviewers: When I rewrote the procedures, I used the labels on UI elements for the upcoming [Semgrep App PR 9426](https://github.com/returntocorp/semgrep-app/pull/9426) merge.

I also went through a bunch of files, so I cleaned up some stuff:

~~- "Ignoring files, folders, and code" was renamed to "Ignoring files and folders", because the `nosemgrep` annotation actually generates a finding and then ignores it as a triage action, as opposed to "excluding file from scan" which is what Semgrep does for files and folders. The distinction was added to a callout box in Ignoring files and folders but we never updated the title.~~ **I am walking back this decision because I am conflicted about this change because the doc itself provides information on ignoring parts of code through nosemgrep, and it feels natural for the user to search for ignoring code. What I can do is stress the difference between ignoring as skip and ignoring as triage.




Preferred term to refer to Policies should be **Policies page** in most cases or **Policies** if the meaning is clear. ([Definition of page, dialog and other UI elements in Google Style Guide](https://developers.google.com/style/ui-elements#term-windows))

s-santillan commented:
PR on code side: [https://github.com/returntocorp/semgrep-app/pull/9426](https://github.com/returntocorp/semgrep-app/pull/9426)

pabloest reviewed (commented):
Does this page need to be deleted? I saw the redirect for it.

<img width="1624" alt="CleanShot 2023-07-28 at 14 29 56@2x" src="https://github.com/returntocorp/semgrep-docs/assets/302941/59f4971d-5c0b-4d31-977e-cc283cd74fea">

s-santillan commented:
@pabloest Yep good catch I'll remove it!

s-santillan commented:
> Does this page need to be deleted? I saw the redirect for it.
> 

Removed!

pabloest commented:
@s-santillan I think the contents of the PR are good now, however there is one test failing and I’m not sure why.

armchairlinguist commented on docs/semgrep-ci/running-semgrep-ci-without-semgrep-cloud-platform.md:
```diff
@@ -52,8 +52,8 @@ By configuring a job's parameters, you are able to achieve the following goals:
-* Stand-alone CI jobs cannot send [PR or MR comments](/semgrep-cloud-platform/github-pr-comments). These comments describe the finding and help developers resolve vulnerabilities and other code issues.
-* Stand-alone CI jobs cannot block a pull or merge request based on a finding. There is no distinction or tiering between rules. Alternatively, Semgrep Cloud Platform's [Rule Board](/semgrep-code/rule-board) can tier rules and undertake actions, such as blocking PRs or MRs on a finding generated by a reliable rule.
+* Stand-alone CI jobs cannot send [PR comments]((/semgrep-cloud-platform/github-pr-comments) or [MR comments](/semgrep-cloud-platform/gitlab-mr-comments). These comments describe the finding and help developers resolve vulnerabilities and other code issues.
+* Stand-alone CI jobs cannot block a pull or merge request based on a finding. There are no user-defined [rule modes] to distinguish between rules. Alternatively, Semgrep Cloud Platform's [Policies page](/semgrep-code/policies) lets users define certain actions, such as blocking PRs or MRs on a finding generated by a reliable rule.
```
Is there supposed to be a link with the rule modes text?

s-santillan replied:
Good catch, I'll add it in.

armchairlinguist commented on src/components/procedure/_remove-ruleset.mdx:
```diff
@@ -1,20 +1,10 @@
+- When you disable a rule, all previous findings generated by that rule are still retained.
 :::
-
-You can also remove a rule on the Findings page, to do so, follow these steps:
```
This is still possible if you don't use group-by-rule - is there a reason it was removed?

s-santillan replied:
Thanks for catching that. I'll update the docs here :)

armchairlinguist reviewed (commented):
I just had one significant comment - I think some functionality that still exists may have had its instructions removed, unless they just got put somewhere else and I didn't notice.

Also, this change updates names of UI elements, but in the UI the rule board is still "Board". Will this release be coordinated with changing the UI name?

s-santillan commented:
Hi @armchairlinguist ! Here's how I addressed your notes:

> I just had one significant comment - I think some functionality that still exists may have had its instructions removed, unless they just got put somewhere else and I didn't notice.

Yup, added this back in after testing. Thanks for the catch.

> Also, this change updates names of UI elements, but in the UI the rule board is still "Board". Will this release be coordinated with changing the UI name?

Currently In the navbar/sidenav, it's Board; in the page's Header itself, it's called Policies. The code-side PR I was supposed to sync this to is not yet merged with the navbar correction and I'd rather unblock these changes so we can merge now and update the docs with a follow up PR, including correct screenshots.

I updated various endpoints back from /policies to /board for this reason as well.
